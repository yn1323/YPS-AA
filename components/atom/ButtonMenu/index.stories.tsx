import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import Component from '.'
import { Delete } from '@material-ui/icons'

export default {
  title: `atom/${Component.name}`,
  component: Component,
} as ComponentMeta<typeof Component>

const Template: ComponentStory<typeof Component> = args => (
  <Component {...args}>LABEL</Component>
)

export const MenuButton = Template.bind({})
MenuButton.args = {
  icon: <Delete />,
  clickHandler: () => console.log('clicked'),
  showArrowDown: true,
}
