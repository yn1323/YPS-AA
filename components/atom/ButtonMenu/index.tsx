import { Button } from '@material-ui/core'
import { KeyboardArrowDown } from '@material-ui/icons'
import React, { FC } from 'react'

interface Props {
  icon: JSX.Element
  showArrowDown?: boolean
  clickHandler: () => void
}

const ButtonMenu: FC<Props> = ({
  icon,
  showArrowDown = true,
  clickHandler,
  children,
}) => {
  return (
    <Button
      color="inherit"
      startIcon={icon}
      endIcon={showArrowDown ? <KeyboardArrowDown /> : undefined}
      onClick={clickHandler}
    >
      {children}
    </Button>
  )
}

export default ButtonMenu
