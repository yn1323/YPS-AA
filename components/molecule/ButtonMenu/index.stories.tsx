import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import Component from '.'
import { ADMIN_MENU, TIMECARD } from '@constant/layout/menus'

export default {
  title: `molecule/${Component.name}`,
  component: Component,
} as ComponentMeta<typeof Component>

const Template: ComponentStory<typeof Component> = args => (
  <Component {...args}>LABEL</Component>
)

export const NoMenu = Template.bind({})
NoMenu.args = {
  ...TIMECARD[0],
}

export const HasMenu = Template.bind({})
HasMenu.args = {
  ...ADMIN_MENU[0],
}
