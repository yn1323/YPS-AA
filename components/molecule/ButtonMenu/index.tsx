import ListMenu from '@atom/ListMenu'
import { Button } from '@material-ui/core'
import { KeyboardArrowDown } from '@material-ui/icons'
import styled from 'styled-components'
import React, { FC, MouseEvent, useState } from 'react'
import { MenuItem } from '@constant/layout/menus'

interface Props {
  icon: JSX.Element
  items?: MenuItem[]
  delimeterPosition?: number[]
  hasMenu: boolean
  link: string
}

export const TextButton = styled(Button)`
  color: ${({ theme }) => theme.palette.text.secondary};
`

const ButtonMenu: FC<Props> = ({
  icon,
  items = [],
  delimeterPosition,
  hasMenu,
  link,
  children,
}) => {
  const [show, setShow] = useState(false)
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)

  const showMenuHandler = (e: MouseEvent<HTMLElement>) => {
    setShow(true)
    setAnchorEl(e.currentTarget)
  }

  return (
    <>
      <TextButton
        startIcon={icon}
        endIcon={hasMenu ? <KeyboardArrowDown /> : undefined}
        href={link ? link : undefined}
        onClick={!link ? showMenuHandler : undefined}
      >
        {children}
      </TextButton>
      {!!items.length && (
        <ListMenu
          items={items}
          delimeterPosition={delimeterPosition}
          show={show}
          setShow={setShow}
          anchorEl={anchorEl}
        />
      )}
    </>
  )
}

export default ButtonMenu
