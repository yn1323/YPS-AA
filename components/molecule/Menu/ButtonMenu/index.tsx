import { Button } from '@material-ui/core'
import { KeyboardArrowDown } from '@material-ui/icons'
import React, { FC, useRef, useState } from 'react'
import styled from 'styled-components'
import ListMenu from '@atom/ListMenu'
import Popper from '@atom/Popper'
import { MenuItem } from '@constant/layout/menus'

interface Props {
  icon: JSX.Element
  items?: MenuItem[]
  delimeterPosition?: number[]
  hasMenu: boolean
  link: string
}

export const TextButton = styled(Button)`
  color: ${({ theme }) => theme.palette.text.secondary};
`

const ButtonMenu: FC<Props> = ({
  icon,
  items = [],
  delimeterPosition,
  hasMenu,
  link,
  children,
}) => {
  const [show, setShow] = useState(false)
  const ref = useRef(null)

  return (
    <>
      <TextButton
        startIcon={icon}
        endIcon={hasMenu ? <KeyboardArrowDown /> : undefined}
        href={link ? link : undefined}
        onClick={!link ? () => setShow(true) : undefined}
        ref={ref}
      >
        {children}
      </TextButton>
      {!!items.length && (
        <Popper show={show} setShow={setShow} anchorEl={ref.current}>
          <ListMenu items={items} delimeterPosition={delimeterPosition} />
        </Popper>
      )}
    </>
  )
}

export default ButtonMenu
