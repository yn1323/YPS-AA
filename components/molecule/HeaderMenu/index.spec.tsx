import ButtonMenu from '@atom/ButtonMenu'
import TestWrapper from '@test/TestWrapper'
import { mount, render } from 'enzyme'
import React from 'react'
import { act } from 'react-test-renderer'
import Component, { handler1 } from '.'

test(Component.name, async () => {
  let component
  await act(async () => {
    component = await render(
      <TestWrapper>
        <Component isAdmin showTimeCard />
      </TestWrapper>
    )
  })
  expect(component).toMatchSnapshot()
  let component2
  await act(async () => {
    component2 = await render(
      <TestWrapper>
        <Component />
      </TestWrapper>
    )
  })
  expect(component2).toMatchSnapshot()
})

test(`${Component.name}-event`, async () => {
  const component1 = await mount(
    <TestWrapper>
      <Component isAdmin showTimeCard />
    </TestWrapper>
  )

  component1.find(ButtonMenu).map(button => button.simulate('click'))
})
