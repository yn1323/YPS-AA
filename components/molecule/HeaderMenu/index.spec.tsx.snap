// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu 1`] = `
<div
  class="HeaderMenu__ButtonWrapper-sc-37oavk-0 cEMwsH"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
      </span>
      マイページ
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </span>
    </span>
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
      </span>
      申請
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </span>
    </span>
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
          />
        </svg>
      </span>
      タイムカード
    </span>
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      店舗管理
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;

exports[`HeaderMenu 2`] = `
<div
  class="HeaderMenu__ButtonWrapper-sc-37oavk-0 cEMwsH"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
      </span>
      マイページ
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </span>
    </span>
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
      </span>
      申請
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;
